docs/design/reserve-action.md

```diff
@@ -0,0 +1,261 @@
+* Job state is not `Job Ready`
+* Job state is not `Job Condition Ready`
+
+If a job has been created for a long time, but it is still not `running`, it is at risk of starving. If nodes under the cluster
+have reserved resource for the starving job, then there is no need to reserve resources for it again.
+
+#### ForceAllocated
+
+If a node tries to reserve resources for the task under the job, when the node has not reserved resources for other tasks, and 
+the idle resources of the node are larger than the requirement resources of the task, this task can be directly allocated to 
+this node with `ForceAllocated` state.  The `ForceAllocated` state is to avoid subsequent `backfill` action to preempt pod
+resources.
+
+#### FakeAllocated
+
+If the idle resource of the node is smaller than the requirement of task, but the allocated resources of the node is larger than
+the requirement of task, the task can be allocated to this node with `FakeAllocated` state. When a node reserves resources for the 
+task, the state of the task will be set to `FakeAllocated`.
+
+### Node resource reservation rule
+
+#### Node reserve logic
+
+If the allocatable resources of the node is larger than the resource requirement of the task, then the node satisfies the 
+scheduled requirement of the task now or later. This node can reserve resources for this task. If the idle resource of 
+the current node has met the scheduling condition of the task, resources will be allocated to this task, and the task 
+will be directly assigned to the node with `ForceAllocated` state. If the idle resource of the node can not meet the resource 
+requirement of the task, the node will reserve resources for this task, the task will be allocated to this node with 
+`FakeAllocated` state. Once a node reserve resources for a task, no other tasks will be scheduled to this node, for 
+the node's allocatable resources will no longer meet the scheduling requirements of any task.
+
+![reserve-logic](../../docs/images/reserve-logic.png)
+
+A node may reserve resources only for one task, or may reserve resources for multiple tasks, depending on the size of the 
+allocatable resources of node.
+
+#### The proportion of nodes participating in reserve action
+
+Because of that some of node resources was reserved for tasks and will not be used by other tasks, the resources 
+reservation will affect the task scheduling process of the entire cluster. In theory, the more nodes that participate in
+resource reservation, the better it is to solve the problem of the task being starved to death. However, the fewer nodes 
+participating in regular scheduling, and the greater the impact on regular cluster scheduling. Therefore, the number of 
+nodes participating in resource reservation is configurable.
+
+#### Node selection strategy to participate in reserve
+ 
+How to choose which nodes will be selected as candidate nodes for starving job scheduling. The process of selecting a 
```

Comment:
After nodes are reserved for a task, could it be scheduled to other nodes if they have enough resource?

Reply:
In a scheduling round, if the task is reserved for resources, its state will be `ForceAllocated` or `FakeAllocated`. Subsequent actions will only process tasks in the `pending` state, and will not process these tasks. If there is a situation in which the idle resource of the node meets the demand of the task, the `reserve` action needs to ensure that the task is allocated to the node during the `reserve` phase. In the `reserve` phase, nodes with sufficient resources will be selected as candidate nodes. If the node's allocatable resource is larger than the task's required resource, the task will be `ForceAllocated` to the node.

Reply:
How to deal with this situation: In the first reserve pass, node N's idle resource is very low, and its score is very low, so it was not reserved for task T, and we reserved other nodes for task T. However after some time(those scheduler phases have passed again and again), node N has enough idle resource for task T, and those reserved nodes still do not have enough idle resource for it. Will we allocate node N for task T?

Reply:
Reserved node selection is a dynamic process, and the nodes reserved in each scheduling round may be different. When selecting candidate nodes, the score of the nodes is firstly calculated and sorted according to the nodes score, so that the node selected for this task in this time will be better than the node selected in the previous round. As the scheduling round progresses, the final filtered nodes will get better and better until the task resource requirement is met.

Reply:
Get it now, thanks!